samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -323,12 +342,54 @@ public async Task<string> ExtractAudienceAsync(SKContext context)
 
         await this.UpdateResponseStatusOnClient(chatId, "Calculating total token usage");
         this.GetTokenUsages(context, chatContext);
+
         context.Variables.Update(response);
+
         return context;
     }
 
     #region Private
 
+    /// <summary>
+    /// Invokes AzureContentModerator to analyze image content.
+    /// </summary>
+    /// <returns>SKContext containing the image analysis result.</returns>
+    private async Task<SKContext> ProcessImageContent(
+        [Description("The new message")] string message,
+        [Description("Unique and persistent identifier for the user")] string userId,
+        [Description("Name of the user")] string userName,
+        [Description("Unique and persistent identifier for the chat")] string chatId,
+        SKContext chatContext)
+    {
+        var imageCannedResponse = "Great image!";
+        string messageType = "image";
+
+        if (this._contentModerator.ContentModerationOptions!.Enabled)
+        {
+            var moderationResult = await this._contentModerator.ImageAnalysisAsync(message, default);
+            var violationCategories = AzureContentModerator.ParseViolatedCategories(moderationResult, this._contentModerator.ContentModerationOptions.ViolationThreshold);
+
+            if (violationCategories.Count > 0)
+            {
+                await this.SaveNewMessageAsync($"Content is redacted due to potential violation: {string.Join(", ", violationCategories)}", userId, userName, chatId, messageType);
+
+                chatContext.Variables.Update("It seems the content isn't appropriate.");
+            }
+            else
+            {
+                chatContext.Variables.Update(imageCannedResponse);
+                await this.SaveNewMessageAsync(imageCannedResponse, userId, userName, chatId, messageType);
+            }
+        }
+        else
+        {
+            chatContext.Variables.Update(imageCannedResponse);
+            await this.SaveNewMessageAsync(imageCannedResponse, userId, userName, chatId, messageType);
+        }
```

Comment:
nit: you can get rid of the two `else` blocks if you return `chatContext` in the nested `if`:
```suggestion
        if (this._contentModerator.ContentModerationOptions!.Enabled)
        {
            var moderationResult = await this._contentModerator.ImageAnalysisAsync(message, default);
            var violationCategories = AzureContentModerator.ParseViolatedCategories(moderationResult, this._contentModerator.ContentModerationOptions.ViolationThreshold);

            if (violationCategories.Count > 0)
            {
                await this.SaveNewMessageAsync($"Content is redacted due to potential violation: {string.Join(", ", violationCategories)}", userId, userName, chatId, messageType);
                chatContext.Variables.Update("It seems the content isn't appropriate.");
                return chatContext;
            }
        }

        chatContext.Variables.Update(imageCannedResponse);
        await this.SaveNewMessageAsync(imageCannedResponse, userId, userName, chatId, messageType);
        return chatContext;
```

Reply:
This is no longer relevant - treating images as docs only so removed all logic support image as message